docs/configuration.md

# Configuration

Project takes configurations from two places: from config files in `projectRoot/config` folder and from environment variables.

Environment variables takes precedence.

## Loading from files

There is two files, configuration will be loaded from: `default.json` and `${ENV}.json` where `${ENV}` is `ENV` environment variable or `dev` if `ENV` environment variable is not set.

If both files exists they will be merged. `${ENV}.json` takes precedence.

> **_WARNING:_**  `default.json` stored in git so do not put secret information in there (database credentials, passwords, etc.)

## Configuration variables

| Key in file                         | Environment                          | Type   | Required | Default value                                 | Need for                                                                                                                                  |
| ----------------------------------- | ------------------------------------ | ------ | -------- | --------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
| adm.jwt.secret                      | ADM_JWT_SECRET                       | string | false    | admSecret                                     | Секрет для подписи JWT-токенов приложения админки                                                                                         |
| admin.recaptcha.requiredScore       | ADMIN_RECAPTCHA_REQUIRED_SCORE       | float  | false    | 0.7                                           | Требуемый уровень доверия к пользователю                                                                                                  |
| admin.recaptcha.secretKey           | ADMIN_RECAPTCHA_SECRET_KEY           | string | false    |                                               | Секретный токен рекапчи приложения админки                                                                                                |
| admin.ui.url                        | ADMIN_UI_URL                         | string | false    | http://localhost:8080                         | Урл административного интерфейса (без конечного слеша)                                                                                    |
| app.environment                     | APP_ENVIRONMENT                      | string | true     | dev                                           | Название окружения                                                                                                                        |
| app.jwt.secret                      | APP_JWT_SECRET                       | string | false    | appSecret                                     | Секрет для подписи JWT-токенов приложения пользователей                                                                                   |
| app.name                            | APP_NAME                             | string | true     | sport                                         | Техническое название приложения                                                                                                           |
| app.title                           | APP_TITLE                            | string | false    | Meta Template Base                            | Пользовательское название приложения                                                                                                      |
| app.ui.url                          | APP_UI_URL                           | string | false    | http://localhost:4000                         | Урл интерфейса конечного пользователя (без конечного слеша)                                                                               |
| authorization.backendChecks.enabled | AUTHORIZATION_BACKEND_CHECKS_ENABLED | bool   | false    | false                                         | Проверка авторизации на бекенде включены                                                                                                  |
| bootstrap.enabled                   | BOOTSTRAP_ENABLED                    | bool   | false    | false                                         | Подготавливать окружеине при запуске                                                                                                      |
| checkLoginIframe                    | CHECK_LOGIN_IFRAME                   | bool   | false    | false                                         | Проверка авторизации во фрейме включена                                                                                                   |
| customer.recaptcha.requiredScore    | CUSTOMER_RECAPTCHA_REQUIRED_SCORE    | string | false    | 0.7                                           | Требуемый уровень доверия к пользователю                                                                                                  |
| customer.recaptcha.secretKey        | CUSTOMER_RECAPTCHA_SECRET_KEY        | string | false    |                                               | Секретный токен рекапчи приложения пользователя                                                                                           |
| database.main.migration.uri         | DATABASE_MAIN_MIGRATION_URI          | string | true     | postgresql://postgres:password@localhost:5432 | Строка подключения к основной базе для миграций (должна быть прямой строкой подключения к бд минуя pgbouncer, если он используется)       |
| database.main.readOnly.enabled      | DATABASE_MAIN_READ_ONLY_ENABLED      | bool   | true     | false                                         | Включена ли работа с запросами на чтение через отдельное подключение                                                                      |
| database.main.readOnly.uri          | DATABASE_MAIN_READ_ONLY_URI          | string | false    | postgresql://postgres:password@localhost:5432 | Строка подключения к основной базе только для чтения                                                                                      |
| database.main.write.uri             | DATABASE_MAIN_WRITE_URI              | string | true     | postgresql://postgres:password@localhost:5432 | Строка подключения к основной базе данных для записи                                                                                      |
| es.cloudId                          | ES_CLOUD_ID                          | string | false    |                                               | Идентификатор аккаунта в облачном сервисе ElasticSearch                                                                                   |
| es.enabled                          | ES_ENABLED                           | bool   | false    | false                                         | Эластик включен                                                                                                                           |
| es.node                             | ES_NODE                              | string | false    | http://localhost:9200                         | Нода эластика                                                                                                                             |
| es.password                         | ES_PASSWORD                          | string | false    |                                               | Пароль для авторизации в облачном сервисе ElasticSearch                                                                                   |
| es.tls.rejectUnauthorized           | ES_TLS_REJECT_UNAUTHORIZED           | bool   | false    | false                                         | Запрещать невалидный ssl сертификат                                                                                                       |
| es.username                         | ES_USERNAME                          | string | false    |                                               | Пользователь для авторизации в облачном сервисе ElasticSearch                                                                             |
| exportHtml.url                      | EXPORT_HTML_URL                      | string | false    |                                               | ExportHtml url                                                                                                                            |
| graphql.playground.enabled          | GRAPHQL_PLAYGROUND_ENABLED           | bool   | false    | true                                          | Включение graphql playground (true | false)                                                                                               |
| kafka.auth.enabled                  | KAFKA_AUTH_ENABLED                   | bool   | true     | true                                          | Включена проверка аутентификации                                                                                                          |
| kafka.brokers                       | KAFKA_BROKERS                        | string | false    | localhost:29092;localhost:29094               | Список kafka блокеров                                                                                                                     |
| kafka.enabled                       | KAFKA_ENABLED                        | bool   | false    | false                                         | Кафка включена                                                                                                                            |
| kafka.password                      | KAFKA_PASSWORD                       | string | false    |                                               | Пароль доступа в kafka                                                                                                                    |
| kafka.queue.acks                    | KAFKA_QUEUE_ACKS                     | int    | false    | 1                                             | `-1`(all) все несинхронизированные реплики должны подтвердить (по умолчанию), `0` нет подтверждений, `1` только ждет подтверждения лидера |
| kafka.queue.defaultRetryTime        | KAFKA_QUEUE_DEFAULT_RETRY_TIME       | int    | false    | 20000                                         | Время паузы после первой ошибки, например 20000 мс, потом оно увеличивается экспоненциально с мультипликатором 1.5                        |
| kafka.queue.maxAttemptsSize         | KAFKA_QUEUE_MAX_ATTEMPTS_SIZE        | int    | false    | 10                                            | Максимальное количество попыток обработки ошибки на сообщение                                                                             |
| kafka.queue.stackSize               | KAFKA_QUEUE_STACK_SIZE               | int    | false    | 3                                             | Количество сообщений, обрабатываемых параллельно                                                                                          |
| kafka.queue.supportedVersion        | KAFKA_QUEUE_SUPPORTED_VERSION        | string | false    | 1;2                                           | Поддерживаемые версии сообщения                                                                                                           |
| kafka.queue.waitingInterruptTime    | KAFKA_QUEUE_WAITING_INTERRUPT_TIME   | int    | false    | 60000                                         | Время паузы в очереди ожидания, когда она прошла все сообщения, это чтобы она не крутила сообщения покругу без остановки                  |
| kafka.ssl.enabled                   | KAFKA_SSL_ENABLED                    | bool   | true     | true                                          | Включена работа по ssl                                                                                                                    |
| kafka.ssl.rejectUnauthorized        | KAFKA_SSL_REJECT_UNAUTHORIZED        | bool   | false    | false                                         | Запрещать невалидный ssl сертификат                                                                                                       |
| kafka.username                      | KAFKA_USERNAME                       | string | false    |                                               | Username доступа в kafka                                                                                                                  |
| keycloak.adm.cli.client             | KEYCLOAK_ADM_CLI_CLIENT              | string | false    | backend-api                                   | Клиент для доступа в api keycloak админки                                                                                                 |
| keycloak.adm.cli.secret             | KEYCLOAK_ADM_CLI_SECRET              | string | false    |                                               | Секрет для доступа в api keycloak админки                                                                                                 |
| keycloak.app.cli.client             | KEYCLOAK_APP_CLI_CLIENT              | string | false    | backend-api                                   | Клиент для доступа в api keycloak приложения пользователя                                                                                 |
| keycloak.app.cli.secret             | KEYCLOAK_APP_CLI_SECRET              | string | false    |                                               | Секрет для доступа в api keycloak приложения пользователя                                                                                 |
| logs.format                         | LOGS_FORMAT                          | string | false    | plain                                         | Формат логов (plain | json)                                                                                                               |
| loki.url                            | LOKI_URL                             | string | false    |                                               | Урл для доступа в Loki. Используется для запроса бизнес-логов                                                                             |
| oidc.adm.issuer                     | OIDC_ADM_ISSUER                      | string | false    |                                               | Issuer oidc для админки                                                                                                                   |
| oidc.adm.jwksRejectUnauthorized     | OIDC_ADM_JWKS_REJECT_UNAUTHORIZED    | bool   | false    | false                                         | Запрещать невалидный ssl сертификат                                                                                                       |
| oidc.adm.jwksUri                    | OIDC_ADM_JWKS_URI                    | string | false    |                                               | Uri jwks oidc для проверки авторизации админки                                                                                            |
| oidc.adm.realm                      | OIDC_ADM_REALM                       | string | false    |                                               | Реалм oidc для админки                                                                                                                    |
| oidc.adm.url                        | OIDC_ADM_URL                         | string | false    |                                               | Хост oidc для админки                                                                                                                     |
| oidc.app.realm                      | OIDC_APP_REALM                       | string | false    |                                               | Реалм oidc для приложения пользователя                                                                                                    |
| oidc.app.url                        | OIDC_APP_URL                         | string | false    |                                               | Хост oidc для приложения пользователя                                                                                                     |
| s3.accessKeyId                      | S3_ACCESS_KEY_ID                     | string | true     |                                               | Идентификатор доступа для авторизации в S3                                                                                                |
| s3.bucket.emailFiles                | S3_BUCKET_EMAIL_FILES                | string | true     |                                               | Имя бакета для хранения файлов, отправленных на почту                                                                                     |
| s3.bucket.tmpFilesToDownload        | S3_BUCKET_TMP_FILES_TO_DOWNLOAD      | string | true     |                                               | Имя бакета для хранения временных файлов, созданных для скачивания пользователями                                                         |
| s3.endpoint                         | S3_ENDPOINT                          | string | true     | s3.eu-central-1.wasabisys.com                 | Эндпоинт S3, который использует бекенд                                                                                                    |
| s3.public.endpoint                  | S3_PUBLIC_ENDPOINT                   | string | false    |                                               | Эндпоинт S3, доступный для фронтенда                                                                                                      |
| s3.region                           | S3_REGION                            | string | true     | eu-central-1                                  | Идентификатор доступа для авторизации в S3                                                                                                |
| s3.secretAccessKey                  | S3_SECRET_ACCESS_KEY                 | string | true     |                                               | Секретный ключ для авторизации в S3                                                                                                       |
| sentry.dsn                          | SENTRY_DSN                           | string | false    |                                               | Sentry dsn                                                                                                                                |
| smtp.from                           | SMTP_FROM                            | string | false    |                                               | Почтовый адрес, от имени которого следует отправлять письма                                                                               |
| smtp.host                           | SMTP_HOST                            | string | false    |                                               | Хост почтового сервера                                                                                                                    |
| smtp.pass                           | SMTP_PASS                            | string | false    |                                               | Пароль пользователя для авторизации на почтовом сервере                                                                                   |
| smtp.port                           | SMTP_PORT                            | int    | false    | 465                                           | Порт почтового сервера                                                                                                                    |
| smtp.rejectUnauthorized             | SMTP_REJECT_UNAUTHORIZED             | bool   | false    | false                                         | Реджектить ли невалидные сертификаты                                                                                                      |
| smtp.secure                         | SMTP_SECURE                          | bool   | false    | true                                          | Использовать ли tls                                                                                                                       |
| smtp.user                           | SMTP_USER                            | string | false    |                                               | Имя пользователя для авторизации на почтовом сервере                                                                                      |


docs/localComposeEnvironment.md
### Первый запуск с выполнением инита
```sh
yarn compose:start && ./initLocal.sh && yarn dev
```

### Обычный запуск
```sh
yarn compose:start && yarn dev
```

#### Остановка
```sh
yarn compose:stop
```

## Удаление
```sh
yarn compose:delete
```

## Окружение
Для работы с локально поднятым окружением подготовлен конфиг `local`. Для того, чтобы бекенд и скрипты были направлены на локальные сервисы, команды нужно запускать через `yarn runlify start env=local`, например `yarn runlify start env=local yarn prisma:deploy`

src/config/config.ts
/* eslint-disable max-len */
import {EnvVarConfig} from './types';

// DO NOT EDIT! THIS IS GENERATED FILE

export interface Config {
  env: string;
  admJwtSecret?: string;
  adminRecaptchaRequiredScore?: number;
  adminRecaptchaSecretKey?: string;
  adminUiUrl?: string;
  appEnvironment: string;
  appJwtSecret?: string;
  appName: string;
  appTitle?: string;
  appUiUrl?: string;
  authorizationBackendChecksEnabled?: boolean;
  bootstrapEnabled?: boolean;
  checkLoginIframe?: boolean;
  customerRecaptchaRequiredScore?: string;
  customerRecaptchaSecretKey?: string;
  databaseMainMigrationUri: string;
  databaseMainReadOnlyEnabled: boolean;
  databaseMainReadOnlyUri?: string;
  databaseMainWriteUri: string;
  esCloudId?: string;
  esEnabled?: boolean;
  esNode?: string;
  esPassword?: string;
  esTlsRejectUnauthorized?: boolean;
  esUsername?: string;
  exportHtmlUrl?: string;
  graphqlPlaygroundEnabled?: boolean;
  kafkaAuthEnabled: boolean;
  kafkaBrokers?: string;
  kafkaEnabled?: boolean;
  kafkaPassword?: string;
  kafkaQueueAcks?: number;
  kafkaQueueDefaultRetryTime?: number;
  kafkaQueueMaxAttemptsSize?: number;
  kafkaQueueStackSize?: number;
  kafkaQueueSupportedVersion?: string;
  kafkaQueueWaitingInterruptTime?: number;
  kafkaSslEnabled: boolean;
  kafkaSslRejectUnauthorized?: boolean;
  kafkaUsername?: string;
  keycloakAdmCliClient?: string;
  keycloakAdmCliSecret?: string;
  keycloakAppCliClient?: string;
  keycloakAppCliSecret?: string;
  logsFormat?: string;
  lokiUrl?: string;
  oidcAdmIssuer?: string;
  oidcAdmJwksRejectUnauthorized?: boolean;
  oidcAdmJwksUri?: string;
  oidcAdmRealm?: string;
  oidcAdmUrl?: string;
  oidcAppRealm?: string;
  oidcAppUrl?: string;
  s3AccessKeyId: string;
  s3BucketEmailFiles: string;
  s3BucketTmpFilesToDownload: string;
  s3Endpoint: string;
  s3PublicEndpoint?: string;
  s3Region: string;
  s3SecretAccessKey: string;
  sentryDsn?: string;
  smtpFrom?: string;
  smtpHost?: string;
  smtpPass?: string;
  smtpPort?: number;
  smtpRejectUnauthorized?: boolean;
  smtpSecure?: boolean;
  smtpUser?: string;
}

export const envVarsConfig: EnvVarConfig[] = [{
  id: 'adm.jwt.secret',
  type: 'string',
  title: 'Секрет для подписи JWT-токенов приложения админки',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'admin.recaptcha.requiredScore',
  type: 'float',
  title: 'Требуемый уровень доверия к пользователю',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'admin.recaptcha.secretKey',
  type: 'string',
  title: 'Секретный токен рекапчи приложения админки',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'admin.ui.url',
  type: 'string',
  title: 'Урл административного интерфейса (без конечного слеша)',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'app.environment',
  type: 'string',
  title: 'Название окружения',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 'app.jwt.secret',
  type: 'string',
  title: 'Секрет для подписи JWT-токенов приложения пользователей',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'app.name',
  type: 'string',
  title: 'Техническое название приложения',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 'app.title',
  type: 'string',
  title: 'Пользовательское название приложения',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'app.ui.url',
  type: 'string',
  title: 'Урл интерфейса конечного пользователя (без конечного слеша)',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'authorization.backendChecks.enabled',
  type: 'bool',
  title: 'Проверка авторизации на бекенде включены',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'bootstrap.enabled',
  type: 'bool',
  title: 'Подготавливать окружеине при запуске',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'checkLoginIframe',
  type: 'bool',
  title: 'Проверка авторизации во фрейме включена',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'customer.recaptcha.requiredScore',
  type: 'string',
  title: 'Требуемый уровень доверия к пользователю',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'customer.recaptcha.secretKey',
  type: 'string',
  title: 'Секретный токен рекапчи приложения пользователя',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'database.main.migration.uri',
  type: 'string',
  title: 'Строка подключения к основной базе для миграций (должна быть прямой строкой подключения к бд минуя pgbouncer, если он используется)',
  required: true,
  hidden: true,
  editable: false,
}, {
  id: 'database.main.readOnly.enabled',
  type: 'bool',
  title: 'Включена ли работа с запросами на чтение через отдельное подключение',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 'database.main.readOnly.uri',
  type: 'string',
  title: 'Строка подключения к основной базе только для чтения',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'database.main.write.uri',
  type: 'string',
  title: 'Строка подключения к основной базе данных для записи',
  required: true,
  hidden: true,
  editable: false,
}, {
  id: 'es.cloudId',
  type: 'string',
  title: 'Идентификатор аккаунта в облачном сервисе ElasticSearch',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'es.enabled',
  type: 'bool',
  title: 'Эластик включен',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'es.node',
  type: 'string',
  title: 'Нода эластика',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'es.password',
  type: 'string',
  title: 'Пароль для авторизации в облачном сервисе ElasticSearch',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'es.tls.rejectUnauthorized',
  type: 'bool',
  title: 'Запрещать невалидный ssl сертификат',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'es.username',
  type: 'string',
  title: 'Пользователь для авторизации в облачном сервисе ElasticSearch',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'exportHtml.url',
  type: 'string',
  title: 'ExportHtml url',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'graphql.playground.enabled',
  type: 'bool',
  title: 'Включение graphql playground (true | false)',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.auth.enabled',
  type: 'bool',
  title: 'Включена проверка аутентификации',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.brokers',
  type: 'string',
  title: 'Список kafka блокеров',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.enabled',
  type: 'bool',
  title: 'Кафка включена',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.password',
  type: 'string',
  title: 'Пароль доступа в kafka',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 'kafka.queue.acks',
  type: 'int',
  title: '`-1`(all) все несинхронизированные реплики должны подтвердить (по умолчанию), `0` нет подтверждений, `1` только ждет подтверждения лидера',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.queue.defaultRetryTime',
  type: 'int',
  title: 'Время паузы после первой ошибки, например 20000 мс, потом оно увеличивается экспоненциально с мультипликатором 1.5',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.queue.maxAttemptsSize',
  type: 'int',
  title: 'Максимальное количество попыток обработки ошибки на сообщение',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.queue.stackSize',
  type: 'int',
  title: 'Количество сообщений, обрабатываемых параллельно',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.queue.supportedVersion',
  type: 'string',
  title: 'Поддерживаемые версии сообщения',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.queue.waitingInterruptTime',
  type: 'int',
  title: 'Время паузы в очереди ожидания, когда она прошла все сообщения, это чтобы она не крутила сообщения покругу без остановки ',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.ssl.enabled',
  type: 'bool',
  title: 'Включена работа по ssl',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.ssl.rejectUnauthorized',
  type: 'bool',
  title: 'Запрещать невалидный ssl сертификат',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'kafka.username',
  type: 'string',
  title: 'Username доступа в kafka',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'keycloak.adm.cli.client',
  type: 'string',
  title: 'Клиент для доступа в api keycloak админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'keycloak.adm.cli.secret',
  type: 'string',
  title: 'Секрет для доступа в api keycloak админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'keycloak.app.cli.client',
  type: 'string',
  title: 'Клиент для доступа в api keycloak приложения пользователя',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'keycloak.app.cli.secret',
  type: 'string',
  title: 'Секрет для доступа в api keycloak приложения пользователя',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'logs.format',
  type: 'string',
  title: 'Формат логов (plain | json)',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'loki.url',
  type: 'string',
  title: 'Урл для доступа в Loki. Используется для запроса бизнес-логов',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.adm.issuer',
  type: 'string',
  title: 'Issuer oidc для админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.adm.jwksRejectUnauthorized',
  type: 'bool',
  title: 'Запрещать невалидный ssl сертификат',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.adm.jwksUri',
  type: 'string',
  title: 'Uri jwks oidc для проверки авторизации админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.adm.realm',
  type: 'string',
  title: 'Реалм oidc для админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.adm.url',
  type: 'string',
  title: 'Хост oidc для админки',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.app.realm',
  type: 'string',
  title: 'Реалм oidc для приложения пользователя',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'oidc.app.url',
  type: 'string',
  title: 'Хост oidc для приложения пользователя',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 's3.accessKeyId',
  type: 'string',
  title: 'Идентификатор доступа для авторизации в S3',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 's3.bucket.emailFiles',
  type: 'string',
  title: 'Имя бакета для хранения файлов, отправленных на почту',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 's3.bucket.tmpFilesToDownload',
  type: 'string',
  title: 'Имя бакета для хранения временных файлов, созданных для скачивания пользователями',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 's3.endpoint',
  type: 'string',
  title: 'Эндпоинт S3, который использует бекенд',
  required: true,
  hidden: true,
  editable: false,
}, {
  id: 's3.public.endpoint',
  type: 'string',
  title: 'Эндпоинт S3, доступный для фронтенда',
  required: false,
  hidden: true,
  editable: false,
}, {
  id: 's3.region',
  type: 'string',
  title: 'Идентификатор доступа для авторизации в S3',
  required: true,
  hidden: false,
  editable: true,
}, {
  id: 's3.secretAccessKey',
  type: 'string',
  title: 'Секретный ключ для авторизации в S3',
  required: true,
  hidden: true,
  editable: false,
}, {
  id: 'sentry.dsn',
  type: 'string',
  title: 'Sentry dsn',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.from',
  type: 'string',
  title: 'Почтовый адрес, от имени которого следует отправлять письма',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.host',
  type: 'string',
  title: 'Хост почтового сервера',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.pass',
  type: 'string',
  title: 'Пароль пользователя для авторизации на почтовом сервере',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.port',
  type: 'int',
  title: 'Порт почтового сервера',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.rejectUnauthorized',
  type: 'bool',
  title: 'Реджектить ли невалидные сертификаты',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.secure',
  type: 'bool',
  title: 'Использовать ли tls',
  required: false,
  hidden: false,
  editable: true,
}, {
  id: 'smtp.user',
  type: 'string',
  title: 'Имя пользователя для авторизации на почтовом сервере',
  required: false,
  hidden: false,
  editable: true,
}];
